### PR TITLE
Add Decidable (Contravariant Analogue of Alternative)

### LINKED ISSUE #141: Decidable
**Decidable** is the contravariant version of Alternative.

**Suggested incarnations**
- Comparison (`'a -> 'a -> Lt | Gt | Eq`)
- Equivalence (`'a -> 'a -> bool)` 
- Predicate (`'a -> bool`)

**Links**
- https://hackage.haskell.org/package/contravariant
- https://hackage.haskell.org/package/contravariant-1.5.5/docs/Data-Functor-Contravariant-Divisible.html#g:2
- http://jackkelly.name/blog/archives/2020/08/19/abstracting_over_applicative_alternative_divisible_and_decidable/
- https://www.youtube.com/watch?v=JUVMiRRq6wU

### PR DESCRIPTION
**Pull request by gr-im**
This PR adds some missing infixes operators for Divisible too.

fix #141

### CODE REVIEW

**Review by xvw — DISMISSED** [review]

**Review by xhtmlboi — APPROVED** [review]

File: lib/preface_specs/decidable.mli [hunk 1]
```diff
@@ -0,0 +1,104 @@
+(** [Decidable] is a "Contravariant Alternative". *)
+
+(** {2 Laws}
+
+    To have a predictable behaviour, the instance of [Decidable] must obey some
+    laws.
+
+    + All {!module:Divisible} laws
+    + [choose Either.left x (lose f) = x]
+    + [choose Either.right x (lose f) = x]
```
Comment (d-plaindoux) [inline lib/preface_specs/decidable.mli:10]: Change this law to `choose Either.right (lose f) x = x`

File: lib/preface_specs/decidable.mli [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+module type CORE = WITH_CONTRAMAP_AND_DIVIDE_AND_CONQUER
+(** Basis operations. *)
+
+(** Additional operations. *)
+module type OPERATION = sig
+  type 'a t
+  (** The type held by the [Decidable]. *)
+
+  val lost : Void.t t
+  (** [lose] acting on identity.*)
```
Comment (d-plaindoux) [inline lib/preface_specs/decidable.mli:58]: lose or lost?

File: lib/preface_specs/decidable.mli [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  val chosen : 'a t -> 'b t -> ('a, 'b) Either.t t
+  (** [choose] acting on identity.*)
```
Comment (d-plaindoux) [inline lib/preface_specs/decidable.mli:61]: choose or chosen?

File: test/preface_examples_test/decidable_pp.ml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+  let string = Fun.id
+
+  let const s = Fun.const s
+
+  let of_pp (type a) (module P : PP with type t = a) x =
+    Format.asprintf "%a" P.pp x
+  ;;
+
+  let int = of_pp (module I)
```
Comment (d-plaindoux) [inline test/preface_examples_test/decidable_pp.ml:26]: Oh boy!

File: lib/preface_specs/decidable.mli (hunk 2 above)
Comment (gr-im) [reply lib/preface_specs/decidable.mli:58]: `lose` because `lost` is `lose` acting on identity.

File: lib/preface_specs/decidable.mli (hunk 3 above)
Comment (gr-im) [reply lib/preface_specs/decidable.mli:61]: `choose` because `chosen` is `choose` acting on identity.

File: lib/preface_specs/decidable.mli (hunk 1 above)
Comment (gr-im) [reply lib/preface_specs/decidable.mli]: Thanks!

**Review by d-plaindoux — APPROVED** [review]

**Pull request merged**